src/application/App.tsx

```diff
@@ -36,97 +84,125 @@ const GET_OPERATION_COUNTS: TypedDocumentNode<
 `;
 
 export const App = () => {
+  const mountedRef = useRef(false);
   const [settingsOpen, setSettingsOpen] = useState(false);
   const { data } = useQuery(GET_OPERATION_COUNTS);
+  const [clientNotFoundModalOpen, setClientNotFoundModalOpen] = useState(false);
   const selected = useReactiveVar<Screens>(currentScreen);
-  const reloading = useReactiveVar<boolean>(reloadStatus);
+  const state = useReactiveVar(devtoolsState);
   const [embeddedExplorerIFrame, setEmbeddedExplorerIFrame] =
     useState<HTMLIFrameElement | null>(null);
 
-  // During a reload, reset the current screen to Queries.
   useEffect(() => {
-    if (reloading && selected !== Screens.Queries) {
-      currentScreen(Screens.Queries);
+    let timeout: NodeJS.Timeout;
+    // Don't show connected message on the first render if we are already
+    // connected to the client.
+    if (!mountedRef.current && state === "connected") {
+      return;
     }
-  }, [reloading, selected]);
 
-  if (reloading) {
-    return <div></div>;
-  }
+    if (state === "notFound") {
+      setClientNotFoundModalOpen(true);
+    }
+
+    const dismiss = BannerAlert.show(ALERT_CONFIGS[state]);
```

Comment:
I'm opting to show a banner for the `notFound` state alongside the modal, that way if the user closes the modal, there is still a way for them to retry the connection.